runtime/domain.c

```diff
@@ -276,10 +276,11 @@ static void create_domain(uintnat initial_minor_heap_wsize) {
       goto reallocate_minor_heap_failure;
     }
 
-    domain_state->dls_root = caml_create_root_noexc(Val_unit);
-    if(domain_state->dls_root == NULL) {
+    domain_state->dls_root = Val_unit;
+    if(domain_state->dls_root == (value)NULL) {
```

Comment:
Can't we remove this branch and the label `create_dls_root_failure`?
(`domain_state->dls_root` will always be `Val_unit` and never `NULL`)